# Release 4.1.0

beckfaSICKAG opened the pull request:
# Version 4.1.0

## New features
- Added event based usage of the handleOnNewImageProcessingScores function
### ...

## Improvements

### ...

## Bugfix

### Known issues
- handleOnNewImageProcessingScores crashes on SIM1012 with firmware versions other than [2.1.0, 2.2.0, 2.2.1]

## Following tasks were checked

- [x] Module is coded and structured according to the "Developing guideline for modules" described within the CSK documentation
- [x] All functions/events/parameters are documented within the manifest documentation
- [x] The manifest description of the main CROWN includes main information about the purpose of the module and how to use it in general
- [x] API docu based on the manifest was created and stored within the "docu"-folder of the repository
- [x] Internal LUA code documentation exists for variables and non served functions
- [x] All relevant infos are logged via the SharedLogger 'ModuleLogger'
- [x] Module supports persistent data feature based on 'CSK_Module_PersistentData'
- [x] Module supports user management based on 'CSK_Module_UserManagement'
- [x] No open "ToDos" within the code or at least clearly explained comments why they exist...
- [x] "Version" key in app manifest was updated following semantic versioning (and use '0.x.y' for test / experimental modules which are not yet ready to be officially released)
- [x] Meaningful IDs used for UI elements
- [x] Module was tested on an AppSpace device (at least on SICK AppEngine) with no error message
- [x] README.md is up to date (incl. info of device + firmware the module was tested with)
- [x] CHANGELOG.md is up to date

beckfaSICKAG committed:
# Release 4.1.0

### New features
- Added event based usage of the handleOnNewImageProcessingScores function

beckfaSICKAG committed:
Added a firmware check to prevent crashes on SIM 1012

golluroSICKAG committed:
* Review of PR https://github.com/SICKAppSpaceCodingStarterKit/CSK_Module_MultiDeepLearning/pull/1
- Minor edits

golluroSICKAG commented:
Hi Fabian,

thx for your contribution!
Please check this PR with my suggestions for some minor changes:
https://github.com/beckfaSICKAG/CSK_Module_MultiDeepLearning/pull/1

beckfaSICKAG committed:
Merge pull request #1 from golluroSICKAG/feature/processWithScores

* Review of PR https://github.com/SICKAppSpaceCodingStarterKit/CSK_Mo…

golluroSICKAG committed:
* Fixed link to docu

beckfaSICKAG committed:
Removed handleOnNewImageProcessingScores function

golluroSICKAG committed:
Merge branch 'beckfaSICKAG:feature/processWithScores' into feature/processWithScores

golluroSICKAG committed:
* Edited Changelog regarding function merge
* Adapted input parameter definition for 'processImage' function

beckfaSICKAG committed:
Fixed wrong bindings

beckfaSICKAG committed:
Updated firmware check

golluroSICKAG committed:
Merge branch 'beckfaSICKAG:feature/processWithScores' into feature/processWithScores

golluroSICKAG committed:
* Updated API docu

golluroSICKAG committed:
- Made device/firmware check more generic (to be used in other modules as well...)
- Only check for device + firmware version if it is tried to activate the relevant feature instead of checking in every time during processing

golluroSICKAG committed:
* Updated API docu

golluroSICKAG committed:
* Updated version number to check for

golluroSICKAG commented:
Hi Fabian,
pls check again my edits within this PR:
https://github.com/beckfaSICKAG/CSK_Module_MultiDeepLearning/pull/2
Thx!

beckfaSICKAG committed:
Merge pull request #2 from golluroSICKAG/feature/processWithScores

Feature/process with scores

golluroSICKAG commented:
Merged into repo.
Thx @beckfaSICKAG !
